From 4d7f59c9e60797f4e5717348497f18f80b0bfaef Mon Sep 17 00:00:00 2001
From: hmz007 <hmz007@gmail.com>
Date: Thu, 20 Feb 2020 18:28:31 +0800
Subject: [PATCH 30/73] pwm: rockchip: add shutdown callback

Signed-off-by: hmz007 <hmz007@gmail.com>
---
 drivers/pwm/pwm-rockchip.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/pwm/pwm-rockchip.c b/drivers/pwm/pwm-rockchip.c
index 73352e6fbccb..c2d606d8d832 100644
--- a/drivers/pwm/pwm-rockchip.c
+++ b/drivers/pwm/pwm-rockchip.c
@@ -11,6 +11,7 @@
 #include <linux/module.h>
 #include <linux/of.h>
 #include <linux/of_device.h>
+#include <linux/pinctrl/consumer.h>
 #include <linux/platform_device.h>
 #include <linux/pwm.h>
 #include <linux/time.h>
@@ -404,6 +405,11 @@ static int rockchip_pwm_remove(struct platform_device *pdev)
 	return pwmchip_remove(&pc->chip);
 }
 
+static void rockchip_pwm_shutdown(struct platform_device *pdev)
+{
+	pinctrl_pm_select_sleep_state(&pdev->dev);
+}
+
 static struct platform_driver rockchip_pwm_driver = {
 	.driver = {
 		.name = "rockchip-pwm",
@@ -411,6 +417,7 @@ static struct platform_driver rockchip_pwm_driver = {
 	},
 	.probe = rockchip_pwm_probe,
 	.remove = rockchip_pwm_remove,
+	.shutdown = rockchip_pwm_shutdown,
 };
 module_platform_driver(rockchip_pwm_driver);
 
-- 
2.31.1

