From 75b13aa19dd3ed979e692493e39c1557e976dab7 Mon Sep 17 00:00:00 2001
From: hmz007 <hmz007@gmail.com>
Date: Sat, 29 Aug 2020 17:09:21 +0800
Subject: [PATCH 40/73] usb: dwc3: add support to set irq affinity

Signed-off-by: hmz007 <hmz007@gmail.com>
---
 drivers/usb/dwc3/host.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/usb/dwc3/host.c b/drivers/usb/dwc3/host.c
index 5567ed2cddbe..5cda769fbe1f 100644
--- a/drivers/usb/dwc3/host.c
+++ b/drivers/usb/dwc3/host.c
@@ -48,6 +48,8 @@ int dwc3_host_init(struct dwc3 *dwc)
 	int			ret, irq;
 	struct resource		*res;
 	struct platform_device	*dwc3_pdev = to_platform_device(dwc->dev);
+	struct cpumask		cpumask;
+	u32			cpu_id;
 	int			prop_idx = 0;
 
 	irq = dwc3_host_get_irq(dwc);
@@ -63,6 +65,13 @@ int dwc3_host_init(struct dwc3 *dwc)
 	if (!res)
 		return -ENOMEM;
 
+	if (!of_property_read_u32(dwc->dev->of_node, "handle_cpu_id", &cpu_id)) {
+		cpumask_clear(&cpumask);
+		cpumask_set_cpu(cpu_id, &cpumask);
+		irq_set_affinity(irq, &cpumask);
+		dev_dbg(dwc->dev, "setup irq on cpu%d\n", cpu_id);
+	}
+
 	dwc->xhci_resources[1].start = irq;
 	dwc->xhci_resources[1].end = irq;
 	dwc->xhci_resources[1].flags = res->flags;
-- 
2.31.1

