From 0e9b6be99a1c4b8081c1560e369c8d2265a5c761 Mon Sep 17 00:00:00 2001
From: hmz007 <hmz007@gmail.com>
Date: Mon, 9 Nov 2020 14:41:14 +0800
Subject: [PATCH] arm64: dts: rockchip: add PM config for nanopi4

Signed-off-by: hmz007 <hmz007@gmail.com>
---
 .../boot/dts/rockchip/rk3399-nanopi4.dtsi     | 31 +++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-nanopi4.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-nanopi4.dtsi
index 4200ecf0224a..cf4531e9eaff 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-nanopi4.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-nanopi4.dtsi
@@ -13,6 +13,7 @@
 
 /dts-v1/;
 #include <dt-bindings/input/linux-event-codes.h>
+#include <dt-bindings/soc/rockchip,suspend.h>
 #include "rk3399.dtsi"
 #include "rk3399-opp.dtsi"
 
@@ -134,6 +135,36 @@ status_led: led-0 {
 		};
 	};
 
+	pm_config: rockchip-suspend {
+		compatible = "rockchip,pm-rk3399";
+		rockchip,sleep-debug-en = <1>;
+		rockchip,sleep-mode-config = <
+			(0
+			| RKPM_SLP_ARMPD
+			| RKPM_SLP_PERILPPD
+			| RKPM_SLP_DDR_RET
+			| RKPM_SLP_PLLPD
+			| RKPM_SLP_CENTER_PD
+			| RKPM_SLP_AP_PWROFF
+			)
+			>;
+		rockchip,wakeup-config = <
+			(0
+			| RKPM_GPIO_WKUP_EN
+			| RKPM_PWM_WKUP_EN
+			)
+			>;
+		rockchip,pwm-regulator-config = <
+			(0
+			| PWM2_REGULATOR_EN
+			)
+			>;
+		rockchip,power-ctrl =
+			<&gpio1 17 GPIO_ACTIVE_HIGH>,
+			<&gpio1 14 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+
 	sdio_pwrseq: sdio-pwrseq {
 		compatible = "mmc-pwrseq-simple";
 		clocks = <&rk808 1>;
-- 
2.31.1

